test_abnormality_detection.py

```diff
@@ -30,23 +33,30 @@ def parse_args() -> argparse.Namespace:
 
 if __name__ == "__main__":
     args = parse_args()
-
     # select type of device
     device = torch.device('cuda') if args.gpu else torch.device('cpu')
 
     # load grasping model
-    grasping_model = abnormality_detection.load_model(Path(args.model_path), device)
-
-    # load input image from argument (converted from RGBA to RGB)
+    model = abnormality_detection.load_model(Path(args.model_path), device)
+    # load input image (convert from RGBA to RGB)
     image = Image.open(args.image_path).convert('RGB')
-
-    # call grasping inference and print what's been returned
-    detected_objects, visualization_results = abnormality_detection.grasping_inference(
-        grasping_model, image, device)
-    print(detected_objects)
-    visualization_results.show()
-    # OR TODO passing image and allowed_regions as arguments for judge_image()
-
-    # for object in detected_objects:
-    #     # TODO if object is outside allowed region, print its information
-    #     pass
+    # load allowed_regions map
+    map_ = Image.open(args.allowed_regions).convert('L')
+    allowed_regions = np.asarray(map_)
+    allowed_regions = allowed_regions.T
+
+    # judge image according to allowed_regions
+    detected_items, visualization = abnormality_detection.judge_image(
+        model, image, allowed_regions, device)
+
+    # save visualization_results under format inputImagePath_timestamp.jpg
+    dt = datetime.now()
+    ts = str(datetime.timestamp(dt))
+    no_extension = os.path.splitext(args.image_path)[0]
+    # visualization_results.show()
+    visualization.save(f"{no_extension}_{ts}.jpg")
+
+    for object in detected_items:
+        #  if object is outside allowed region, print its information
+        if object["judge"] == False:
```

Comment:
We need to fix this key => `inside_allowed_region`

```suggestion
        if object["inside_allowed_region"] == False:
```